### PR TITLE
Update _TimeViewer's plotter only when necessary

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR updates the render window only when it is expected or necessary (meaning after a user interaction) instead of being called constantly at a very small interval. I expect this to lower the burden on the CPU.

**Note**: My CPU usage for a standard script went from ~25-30% to around 1% on idle.

It's an item of #7162

### CODE REVIEW

**Comment by GuillaumeFavelier** [comment]
Can you try this @hoechenberger ? It would be nice if this could at least improve your situation.

File: mne/viz/_brain/_timeviewer.py [hunk 1]
```diff
@@ -416,7 +425,6 @@ def play(self):
             self.time_call(idx, update_widget=True)
             if time_point == max_time:
                 self.playback = False
-        self.plotter.update()  # critical for smooth animation
```
Comment (larsoner) [inline mne/viz/_brain/_timeviewer.py]: This always should have been indented one more level :( But if you now have `time_call` effectively trigger the `.update()`, then indeed it makes sense to remove it entirely now

Comment (GuillaumeFavelier) [reply mne/viz/_brain/_timeviewer.py]: It was necessary after PyVista 0.24, probably something can be improved upstream

**Comment by hoechenberger** [comment]
@GuillaumeFavelier I tested this with:

```python
In [1]: %pwd
Out[1]: '/Users/hoechenberger/Development/mne-python/examples/inverse'

In [2]: import mne

In [3]: mne.viz.set_3d_backend('pyvista')
Using pyvista 3d backend.


In [4]: %run plot_vector_mne_solution.py
```
And I get a mayavi window… I _think_ there might be a PyVista window shortly popping up and then disappearing, but I'm not sure (maybe it's the mayavi window all along, changing some of its properties)

**Comment by GuillaumeFavelier** [comment]
@hoechenberger https://github.com/mne-tools/mne-python/pull/7060 is not merged yet

**Comment by agramfort** [comment]
can you make sure you have pyvista >= 0.24?

>

**Comment by larsoner** [comment]
The vector PR has not been merged, you need to test a standard STC for this PR

**Comment by hoechenberger** [comment]
Oh sorry, I thought it had been merged. Losing track with all these PRs and amazing new features by @GuillaumeFavelier :D

This PR fixes the CPU hogging issue for me!

**Comment by GuillaumeFavelier** [comment]
> This PR fixes the CPU hogging issue for me!

Fantastic!

**Comment by larsoner** [comment]
@GuillaumeFavelier can you fix the conflict? then I'll merge

**Comment by GuillaumeFavelier** [comment]
I'm on it

**Comment by larsoner** [comment]
CIs were happy before merge and still are locally so I'll go ahead and merge, thanks @GuillaumeFavelier

**Comment by GuillaumeFavelier** [comment]
I'll check how it goes directly on `master` then

**Pull request merged**